### DISCUSSION TITLE
Bmp180 timeout

### DISCUSSION THREAD

**Discussion by RED-124** [micropython/micropython#9673] [General]
Hello guys (and girls),

I've been working on a little project recently using both an mpu6050 and a bmp180 each on an I2C bus (i know, kind of strange but ultimately both sensors will have redundency, one of each on each bus)

My problem is that on about 150 try, the code worked about 10 times, it apears the program pretty much always stops on the bmp180 initialisation with a "OSError: [Errno 110] ETIMEDOUT"

I thought it might just be the board failing to sinc time with the sensor on the first try, so i made a "try/except:continue" loop on the initialisation of the sensor but even after an hour (about 15 000 loop) it still didn't work...

The circuit is a custom pcb so yes, the connections are consistent and working (i checked) with 4.7k pullup resistors to 3.3V
If you have any ideas on what could be the problem here, i'd be very glad to ear them... I've been stuck on the same problem for a week now and i really don't want to have to reorder new pcb's and different new sensors...

(Btw, i may seem strange that i'm using the mpu6050 library's output as a list but i modified said library to allow it)

[Code.txt](https://github.com/micropython/micropython/files/9813953/Code.txt)

**Comment by robert-hh** [comment]
Which board & port do you use, and which firmware version?

**Reply by RED-124** [reply]
I'm runing a PyBoard V1.0 without accelerometer on the latest firmware there is

**Reply by robert-hh** [reply]
So the bmp180 driver fails. Which one do you use? 
I cannot replicate that here exactly, because I have a PyBoard v1.1 and a BMP085, which works fine.

**Reply by robert-hh** [reply]
Does changing the I2C baudrate help, like to a higher rate, like 100000?

**Reply by RED-124** [reply]
Alright, just tried it with 10 000,100 000 and 500 000, still nothing (and the sensor is a gy-68)

**Reply by robert-hh** [reply]
which bmp180.py script do you use?

**Reply by RED-124** [reply]
I used this one (the error is on "self._bmp_i2c.start()" at line 20)
[bmp180.txt](https://github.com/micropython/micropython/files/9824437/bmp180.txt)

**Reply by robert-hh** [reply]
Please delete that line. It is not required, and in newer versions of the firmware is will cause an error to be raised.

**Reply by RED-124** [reply]
alright, now i have an error on the next line: "self.chip_id = self._bmp_i2c.readfrom_mem(_bmp_addr, 0xD0, 2)"

**Reply by robert-hh** [reply]
What does the error message tell?

**Reply by RED-124** [reply]
still a timeout error

**Reply by robert-hh** [reply]
If try that with your set-up, I jest get a warning:
`Warning: I2C(-1, ...) is deprecated, use SoftI2C(...) instead`
Which is printed since your code is using SoftI2C. Besides that, it works here with the PyBoard v1.1 and a BMP085 or BMP180. Different to my first note, I had a BMP180 here, but it is kind of broken or was never calibrated. The data it returns is wrong.

**Reply by robert-hh** [reply]
Then either the wiring is not right, or the BMP180 is broken. Are you sure that the pull.up resistors are in place and are 4.7 kOhm? GND connected? Do you have any tool to visualize the data timing & levels?

**Reply by robert-hh** [reply]
Recent firmware, although I2C is stable since long for STM32.

**Reply by RED-124** [reply]
mhhh, the pullups are 10k, which should work according to the datasheet (between 10k and 2.2k) but i can try changing them for lower value ones...
And sadly my oscilloscope has been borrowed so i'll only have access to it at the end of next week

**Reply by robert-hh** [reply]
10k is fine as well with a short wire.

**Reply by RED-124** [reply]
I was getting kinda annoyed by this so i just tied random things and after 30+ try, i can confidently say that the bmp085 library works just fine with the bmp180

**Reply by robert-hh** [reply]
That sounds good. I tested with both libraries, and both worked for me, even if I naturally prefer the bmp085 lib. I compared both and did not see any substantial difference for the basic I2C communication.

**Comment by octaprog7** [comment]
Try adding another zero to freq!
BAR = I2C(scl=Pin('Y9'), sda=Pin('Y10'), freq=100_000)
